### ISSUE TITLE
incompatable with asyncio.run

### ISSUE THREAD

**Issue by graingert** [jaraco/keyring#396]
**Describe the bug**

keyring stops working after using asyncio.run

**To Reproduce**
```
Python 3.8.0a3 (default, Apr  3 2019, 07:37:42) 
[GCC 8.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import keyring
>>> keyring
<module 'keyring' from '/home/username/.virtualenvs/redacted/lib/python3.8/site-packages/keyring/__init__.py'>
>>> keyring.set_password("example", "example", "example")
>>> keyring.get_password("example", "example")
'example'
>>> async def amain():
...     raise Exception("failure")
... 
>>> import asyncio
>>> asyncio.run(amain())
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/lib/python3.8/asyncio/runners.py", line 43, in run
    return loop.run_until_complete(main)
  File "/usr/lib/python3.8/asyncio/base_events.py", line 589, in run_until_complete
    return future.result()
  File "<stdin>", line 2, in amain
Exception: failure
>>> keyring.get_password("example", "example")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/username/.virtualenvs/redacted/lib/python3.8/site-packages/keyring/core.py", line 55, in get_password
    return _keyring_backend.get_password(service_name, username)
  File "/home/username/.virtualenvs/redacted/lib/python3.8/site-packages/keyring/backends/SecretService.py", line 67, in get_password
    collection = self.get_preferred_collection()
  File "/home/username/.virtualenvs/redacted/lib/python3.8/site-packages/keyring/backends/SecretService.py", line 50, in get_preferred_collection
    bus = secretstorage.dbus_init()
  File "/home/username/.virtualenvs/redacted/lib/python3.8/site-packages/secretstorage/__init__.py", line 37, in dbus_init
    connection = connect_and_authenticate()
  File "/home/username/.virtualenvs/redacted/lib/python3.8/site-packages/jeepney/integrate/blocking.py", line 82, in connect_and_authenticate
    conn = DBusConnection(sock)
  File "/home/username/.virtualenvs/redacted/lib/python3.8/site-packages/jeepney/integrate/blocking.py", line 20, in __init__
    hello_reply = self.bus_proxy.Hello()
  File "/home/username/.virtualenvs/redacted/lib/python3.8/site-packages/jeepney/integrate/blocking.py", line 64, in inner
    return self._connection.send_and_get_reply(msg)
  File "/home/username/.virtualenvs/redacted/lib/python3.8/site-packages/jeepney/integrate/blocking.py", line 45, in send_and_get_reply
    future = self.send_message(message)
  File "/home/username/.virtualenvs/redacted/lib/python3.8/site-packages/jeepney/integrate/blocking.py", line 24, in send_message
    future = self.router.outgoing(message)
  File "/home/username/.virtualenvs/redacted/lib/python3.8/site-packages/jeepney/routing.py", line 26, in outgoing
    self.awaiting_reply[msg.header.serial] = handle = self.handle_factory()
  File "/usr/lib/python3.8/asyncio/events.py", line 636, in get_event_loop
    raise RuntimeError('There is no current event loop in thread %r.'
RuntimeError: There is no current event loop in thread 'MainThread'.
>>> import pkg_resources
>>> pkg_resources.get_distribution("keyring")
keyring 19.0.2 (/home/username/.virtualenvs/redacted/lib/python3.8/site-packages)
```

**Expected behavior**

No stack trace

**Environment**

- OS: [e.g. macOS]
```
$ lsb_release -a
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 19.04
Release:	19.04
Codename:	disco
```

```
$ pip list | grep keyring
keyring 19.0.2
$ keyring --list-backends
keyrings.alt.file.EncryptedKeyring (priority: 0.6)
keyring.backends.fail.Keyring (priority: 0)
keyring.backends.SecretService.Keyring (priority: 5)
keyrings.alt.file.PlaintextKeyring (priority: 0.5)
keyring.backends.chainer.ChainerBackend (priority: 10)
```

**Additional context**

**Comment by graingert** [comment]
moved: https://github.com/jaraco/keyring/issues/397#issuecomment-519946430

**Comment by jaraco** [comment]
I'm a little confused. Does every project in PyPI need to be rewritten for asyncio? If so, that doesn't seem sustainable, and duplicating every method that might ultimately have IO interaction to have an async doppleganger seems unruly and messy. Keyring aims to supply a small and simple interface with `get/set/delete_password`. Adding async variants to each of those will double the required interfaces and will likely require each and every backend to implement those variants. I'm struggling to see how that's viable.

Perhaps the `aget_keyring()` approach may be viable without proliferating interfaces, though I haven't done enough async programming to know with confidence.

Would you be willing to explore an adaptation of the Keyring base class that doesn't impose any necessary changes on the backends?

**Comment by takluyver** [comment]
This should be fixed with the new version of jeepney I just released.

**Comment by mitya57** [comment]
This is fixed with jeepney ≥ 0.4.1, so closing this issue.
